### PR TITLE
Add setitem method for echodata

### PR DESCRIPTION
**Pull request by lsetiawan**
This PR adds `setitem` per @b-reyes request for overriding the underlying dataset.

### CODE REVIEW

File: echopype/echodata/echodata.py [hunk 1]
```diff
@@ -188,14 +188,27 @@ def __get_dataset(self, node: DataTree) -> Optional[xr.Dataset]:
...
+
     def __getitem__(self, __key: Optional[str]) -> Optional[xr.Dataset]:
         if self._tree:
             try:
-                if __key in ["Top-level", "/"]:
-                    # Access to root
-                    node = self._tree
-                else:
-                    node = self._tree[__key]
+                node = self.__get_node(self, __key)
```
Comment (b-reyes) [inline echopype/echodata/echodata.py:200]:
```suggestion
                node = self.__get_node(__key)
```

I think `self ` should be removed? Unless this is like the absolute import situation?

File: echopype/echodata/echodata.py [hunk 2]
```diff
@@ -188,14 +188,27 @@ def __get_dataset(self, node: DataTree) -> Optional[xr.Dataset]:
...
+                return self.__get_dataset(node)
+            except ChildResolverError:
+                raise GroupNotFoundError(__key)
+        else:
+            raise ValueError("Datatree not found!")
+
+    def __setitem__(self, __key: Optional[str], __newvalue: Any) -> Optional[xr.Dataset]:
+        if self._tree:
+            try:
+                node = self.__get_node(self, __key)
```
Comment (b-reyes) [inline echopype/echodata/echodata.py:210]:
```suggestion
                node = self.__get_node(__key)
```

File: echopype/echodata/echodata.py [hunk 3]
```diff
@@ -607,7 +620,12 @@ def mapping_search_variable(mapping, keys, default=None):
                     "time1",
                     mapping_search_variable(
                         extra_platform_data,
-                        ["heave", "HEAVE", "vertical_offset", "VERTICAL_OFFSET"],
+                        [
+                            "heave",
+                            "HEAVE",
+                            "vertical_offset",
+                            "VERTICAL_OFFSET",
+                        ],
```
Comment (b-reyes) [inline echopype/echodata/echodata.py:628]: This just must be because of `pre-commit run`? Nothing seems to be changed.

Comment (lsetiawan) [reply echopype/echodata/echodata.py:628]: Yepp pre-commit.

**Comment by b-reyes** [comment]
@lsetiawan thank you for implementing this! I tried it out and it works great. I think this is ready to merge.

**Pull request merged**